sdk/iothub/azure-resourcemanager-iothub/src/test/java/com/azure/resourcemanager/iothub/generated/IotHubManagerTests.java

```diff
@@ -0,0 +1,161 @@
+import com.azure.resourcemanager.iothub.models.FallbackRouteProperties;
+import com.azure.resourcemanager.iothub.models.FeedbackProperties;
+import com.azure.resourcemanager.iothub.models.IotHubDescription;
+import com.azure.resourcemanager.iothub.models.IotHubProperties;
+import com.azure.resourcemanager.iothub.models.IotHubSku;
+import com.azure.resourcemanager.iothub.models.IotHubSkuInfo;
+import com.azure.resourcemanager.iothub.models.MessagingEndpointProperties;
+import com.azure.resourcemanager.iothub.models.RoutingProperties;
+import com.azure.resourcemanager.iothub.models.RoutingSource;
+import com.azure.resourcemanager.iothub.models.ResourceIdentityType;
+import com.azure.resourcemanager.iothub.models.StorageEndpointProperties;
+import com.azure.resourcemanager.resources.ResourceManager;
+import io.netty.util.internal.StringUtil;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Random;
+
+public class IotHubManagerTests extends TestBase {
+
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.US_WEST2;
+    private String resourceGroupName = "rg" + randomPadding();
+    private IotHubManager iotHubManager;
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        iotHubManager = IotHubManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups()
+                .define(resourceGroupName)
+                .withRegion(REGION)
+                .create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testIotHubDescription() {
+        IotHubDescription iotHubDescription = null;
+        try {
+            String descriptionName = "iotHub" + randomPadding();
```

Comment:
nit, why it is call `descriptionName`, instead of e.g. `hubName`?

Reply:
Fixed in the new version.